### ISSUE TITLE
Can not access *.py files for execution

### ISSUE THREAD

**Issue by iva-nova-e-katerina** [Significant-Gravitas/Auto-GPT#1896]
### Duplicates

- [X] I have searched the existing issues

### Steps to reproduce 🕹

...
EXECUTE_LOCAL_COMMANDS=True
...
python3 scripts/main.py --continuous --gpt3only
y

### Current behavior 😯

```
NEXT ACTION:  COMMAND = execute_python_file ARGUMENTS = {'file': 'evaluate_net_worth_strategies.py'}
Executing file 'evaluate_net_worth_strategies.py' in workspace 'auto_gpt_workspace'
SYSTEM:  Command execute_python_file returned: Error: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))

```

But 'evaluate_net_worth_strategies.py exists! in Auto-GPT/auto_gpt_workspace ! I swear

### Expected behavior 🤔

*.py file execution

### Your prompt 📝

```yaml
# Paste your prompt here
```

**Comment by Pwuts** [comment]
To me it seems that it is executing the file, which throws the `Error while fetching server API version`.

**Comment by iva-nova-e-katerina** [comment]
it can't access the disk :(

```
NEXT ACTION:  COMMAND = execute_shell ARGUMENTS = {'command_line': 'sudo mkdir /home/user/Auto-GPT/autogpt/auto_gpt_workspace'}
SYSTEM:  Command execute_shell returned: Error: [Errno 2] No such file or directory: '/home/user/Auto-GPT/autogpt/auto_gpt_workspace'
 THOUGHTS:  The 'autogpt' directory seems to be missing. Before we can create a 'auto_gpt_workspace' under it, we will have to create 'autogpt'. Let me create all of them using the 'execute_shell' command
REASONING:  Creating the directories in the correct sequence ensures that they are created at the right location and can aid in avoiding errors from the permission system.
PLAN: 
-  Create the directory 'autogpt' using 'sudo mkdir /home/user/Auto-GPT/autogpt'.\\n- Create the directory 'auto_gpt_workspace' using the command 'sudo mkdir /home/user/Auto-GPT/autogpt/auto_gpt_workspace'
CRITICISM:  I must exercise caution while using the 'sudo' command to mitigate any potential risks and ensure the proper functioning and security of the system.
SPEAK:  Let me create the necessary directories using 'sudo mkdir' commands arranged in the correct sequence.
Attempting to fix JSON by finding outermost brackets 
Apparently json was fixed. 
NEXT ACTION:  COMMAND = execute_shell ARGUMENTS = {'command_line': 'sudo mkdir /home/user/Auto-GPT/autogpt && sudo mkdir /home/user/Auto-GPT/autogpt/auto_gpt_workspace'}
SYSTEM:  Command execute_shell returned: Error: [Errno 2] No such file or directory: '/home/user/Auto-GPT/autogpt/auto_gpt_workspace'

```

> To me it seems that it is executing the file, which throws the `Error while fetching server API version`.

**Comment by Pwuts** [comment]
That is an unrelated issue, duplicate of #1862.

**Comment by Pwuts** [comment]
It should be at least partially fixed now. Can you pull and try again?

**Comment by iva-nova-e-katerina** [comment]
> It should be at least partially fixed now. Can you pull and try again?
`git pull`


Bug is still here 

```

NEXT ACTION:  COMMAND = execute_python_file ARGUMENTS = {'file': 'evaluate_recommendations.py'}
Executing file 'evaluate_recommendations.py' in workspace '/home/user/Auto-GPT/auto_gpt_workspace'
SYSTEM:  Command execute_python_file returned: Error: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
```

**Comment by adrianlzt** [comment]
Looks like the execution differs if you are running autogpt inside of a docker container.

That error ``Error while fetching server API version`` looks like is being returned by python trying to talk with the docker daemon (do you have docker running and accessible to your user?).

A, maybe risky, workaround is to create the file ``/.dockerenv``, used by autogpt to [check if it is running inside of docker](https://github.com/Significant-Gravitas/Auto-GPT/blob/f41febd3ae9242961dc7e940916cd37255abf7f7/autogpt/commands/execute_code.py#L117).

If that's true, it will just try to execute the python script, without using docker.

**Comment by Stunspot** [comment]
The issue is that someone rewrote the execute_python_file command to assume it was in a docker container. An app expecting to be in a docker container that isn't throws a file not found error. Was going nuts from this. Just now had ChatGPT rip out the docker crap for me. Here's my execute_code.py file:
[execute_code - execute_python_file fixed.py.txt](https://github.com/Significant-Gravitas/Auto-GPT/files/11250293/execute_code.-.execute_python_file.fixed.py.txt)

**Comment by Jude-a11y** [comment]
> The issue is that someone rewrote the execute_python_file command to assume it was in a docker container. An app expecting to be in a docker container that isn't throws a file not found error. Was going nuts from this. Just now had ChatGPT rip out the docker crap for me. Here's my execute_code.py file:
> [execute_code - execute_python_file fixed.py.txt](https://github.com/Significant-Gravitas/Auto-GPT/files/11250293/execute_code.-.execute_python_file.fixed.py.txt)

It seems right, but after I changed the execute_code.py file as your did and ran the command, the return of the SYSTEM was nothing, while it should return the output of the Python program.

**Comment by Stunspot** [comment]
I'm sorry to hear that. I just now ran it through some code analysis tools and they all say it's good. First thing I did after replacing it was have the bot write and execute a hello world. Then I had it generate and execute a "moderately complicated python file". It's happily pythoning away as we speak,.

**Comment by johnbrisbin** [comment]
> The issue is that someone rewrote the execute_python_file command to assume it was in a docker container. An app expecting to be in a docker container that isn't throws a file not found error. Was going nuts from this. Just now had ChatGPT rip out the docker crap for me. Here's my execute_code.py file: [execute_code - execute_python_file fixed.py.txt](https://github.com/Significant-Gravitas/Auto-GPT/files/11250293/execute_code.-.execute_python_file.fixed.py.txt)
Works for me. Thanks.

**Comment by Stunspot** [comment]
Glad my little ChatGPT toolset can help people.

**Comment by Pwuts** [comment]
Superseded by #2006

**Comment by iva-nova-e-katerina** [comment]
bug is still here 

```
NEXT ACTION:  COMMAND = execute_python_file ARGUMENTS = {'file': './calculate_net_worth.py'}
Executing file './calculate_net_worth.py' in workspace '/home/user/Auto-GPT/auto_gpt_workspace'
SYSTEM:  Command execute_python_file returned: Error: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
```


```
NEXT ACTION:  COMMAND = execute_python_file ARGUMENTS = {'file': 'calculate_net_worth.py'}
Executing file 'calculate_net_worth.py' in workspace '/home/user/Auto-GPT/auto_gpt_workspace'
SYSTEM:  Command execute_python_file returned: Error: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))

```

**Comment by Pwuts** [comment]
@iva-nova-e-katerina please continue in #2006 if you still have issues
